lib/BridgedRoom.js

```diff
@@ -91,6 +91,18 @@ BridgedRoom.prototype.unlinkMatrixRoom = function(matrixRoomId) {
     ]);
 };
 
+BridgedRoom.prototype.swapRoomIds = function(oldRoomId, newRoomId) {
+    if (this._portalMatrixRoomId === oldRoomId) {
+        this.setPortalMatrixRoomId(newRoomId);
+    }
+
+    const index = this._linkedMatrixRoomIds.indexOf(oldRoomId);
+    if (index) {
+        this._linkedMatrixRoomIds.splice(index, 1);
+        this.linkMatrixRoom(newRoomId);
```

Comment:
The relationship between `linkMatrixRoom()` and `setPortalMatrixRoomId()` is confusing - seems like they should be wrapped together and the common logic made clearer

Reply:
Yep, I believe one is a canonical roomId for portal rooms and the other one is for plumbed. Agreed these should not be run across different functions, but it feels like a problem I want to solve in another PR.